engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/GetAtomFieldsNode.java

```diff
@@ -15,14 +19,32 @@
     description = "Gets the fields of an unresolved atom.",
     autoRegister = false)
 public abstract class GetAtomFieldsNode extends Node {
+  private static final String LIMIT = "3";
+
   static GetAtomFieldsNode build() {
     return GetAtomFieldsNodeGen.create();
   }
 
   abstract EnsoObject execute(Atom atom);
 
+  @Specialization(
+      limit = LIMIT,
+      guards = {"atom.getConstructor() == cons"})
+  @ExplodeLoop
+  EnsoObject doStruct(
+      Atom atom,
+      @Cached("atom.getConstructor()") AtomConstructor cons,
+      @CachedLibrary(limit = LIMIT) StructsLibrary structs) {
+    var arr = new Object[cons.getArity()];
+    for (var i = 0; i < arr.length; i++) {
+      arr[i] = structs.getField(atom, i);
+    }
+    return ArrayLikeHelpers.wrapObjectsWithCheckAt(arr);
+  }
+
   @Specialization
-  EnsoObject doStruct(Atom atom, @CachedLibrary(limit = "2") StructsLibrary structs) {
-    return ArrayLikeHelpers.wrapObjectsWithCheckAt(structs.getFields(atom));
+  @TruffleBoundary
+  final EnsoObject doOther(Atom atom) {
```

Comment:
Question of Truffle experts: If I do just the first `@Specialization(limit = LIMIT, guards = {"atom.getConstructor() == cons"})` then the `execute` method yields `UnsupportedSpecializationException` when we send there more than `LIMIT` of different atom constructors.

I thought `@Fallback` would be appropriate, but DSL processor refuses it. So I ended up with `@Specialization` without any guards behind `@TruffleBoundary` calling the first specialization `@TruffleBoundary`. Can we do better?